Library/Homebrew/download_strategy.rb

```diff
@@ -147,6 +169,16 @@ def fetch
     end
   end
 
+  def fetch_last_commit
+    fetch
+    @local_commit = true
+    last_commit
+  end
+
+  def local_commit?
+    @local_commit
+  end
```

Comment:
I need this method to know If I need to fetch upstream repo or not. It's tightly related to `GitHubGitDownloadStrategy`.

The purpose of `GitHubGitDownloadStrategy` is to easily understand if installed package is up-to-date. If it's not, then we need to fetch it from upstream. So, `local_commit?` shows whether we need to fetch the repo or not. With this method we can call 
`downloader.fetch unless downloader.local_commit?` for any `downloader` descendant of `VCSDownloadStrategy`.

That said, I'm not sure about this behaviour, because probably we need to fetch only if we upgrade the package, but if we do need it, then there arise many cases of how we behave in `outdated` methods. E.g. if we use `GitDownloadStrategy`, we need to fetch the repo to know it's outdated, the same is true for any `VCSDownloadStrategy` except `GitHubGitDownloadStrategy`, so, if we wan't fetch to be performed only when we upgrade package, we need some additional checks for the `downloader` to be `GitHubGitDownloadStrategy` (and I'm not sure where we put them). Thus, I made `local_commit?` method for now and used `VCSDownloadStrategy` interface.